kmir/src/tests/integration/data/prove-rs/bitwise-shift.rs

```diff
@@ -0,0 +1,43 @@
+fn main() {
+    // Shl basic
+    assert!(1_u8 << 3_u8 == 8);
+    assert!(1_u8 << 3_i8 == 8); // Negative on RHS if in 0..size
+    assert!(1_i8 << 3_u8 == 8);
+    assert!(1_i8 << 3_i8 == 8);
+
+    // Shl Overflow
+    assert!(255_u8 << 1_u8 == 254);   
+    assert!(-127_i8 << 1_u8 == 2);   
```

Comment:
See the above comment on `onShift`, the added test will probably fail.

```suggestion
    // Shl Overflow
    assert!(255_u8 << 1_u8 == 254);   
    assert!(-127_i8 << 1_u8 == 2);   
    assert!(1_i8 << 7_u8 == -128);
```

Reply:
It didn't, but it should of! I uncovered another bug figuring out why!